index.js

```diff
@@ -73,7 +72,7 @@ function interpolate(t, order, points, knots, weights, result) {
   // convert back to cartesian and return
   var result = result || [];
   for(i=0; i<d; i++) {
-    result[i] = v[s][i] / v[s][d];
+    result[i] = v[s][i] / weights[s];
```

Comment:
`v[s][d]` is actually a pyramid value, not the original untouched `v[i][j] = weights[i]` value. This change breaks the weight-boosted test case.